common/src/main/java/com/mx/path/core/common/reflection/Annotations.java

```diff
@@ -129,16 +131,28 @@ public final int getPosition() {
     public final void setPosition(int position) {
       this.position = position;
     }
+  }
 
+  /**
+   * Get all fields with their annotations. This is deprecated as the name indicates filtering.
+   *
+   * @param klass
+   * @return a list of FieldWithAnnotation objects
+   * @deprecated Use {@link Annotations#fieldsAndAnnotations(Class)}
+   */
+  @Deprecated
+  public static List<FieldWithAnnotations> fieldsWithAnnotations(Class<?> klass) {
+    return fieldsAndAnnotations(klass);
   }
```

Comment:
I'm a little confused with the rename and deprecation. It seems you are not only deprecating this, but changing what it does. Isn't this going to get ancestor fields now as well? Is that intentional?

Reply:
I'm renaming this because we currently have
fieldsWithAnnotations(class) - returns ALL fields (including their annotations)
fieldsWithAnnotation(annotation, class) - returns a filtered list of fields that are annotated by 'annotation'

For a long time I thought the first method was only returning fields that had 'any' annotation.

We're using 'With' to mean two different things.

fieldsWithAnnotations(class) is being renamed to fieldsAndAnnotations(class).

I'm 'adding' (deprecated) fieldsWithAnnotations(class) for backward compatibility. I thought about not adding this. I'd be down to just do the rename. I don't think this is used anywhere else than where I've changed in this PR.

YES, the other change is that fieldsAndAnnotations now returns ancestor fields, which is really the point of this PR.